### DISCUSSION TITLE
How about Structure Coefficients?

### DISCUSSION THREAD

**Discussion by ericmelse** [jluchman/domin#14] [Ideas]
Dear Joseph,

Not so much an issue I have but a question that actually is an option request.
I assume that you are familiar with the paper 'Tools to support interpreting multiple regression in the face of multicollinearity' published in _[Frontiers of Psychology](https://doi.org/10.3389/fpsyg.2012.00044)_ by Kraha, A., Turner, H., Nimon, K., Zientek, L. R., & Henson, R. K. (2012, 3, 44).
Beside their extensive discussion of dominance analyse they also discuss the so-called Structure Coefficients: 'These are sim-
ply the Pearson correlations between  Y and each predictor. When squared, they yield the proportion of variance in the effect (or, of the  Y scores) that can be accounted for by the predictor alone, irrespective of collinearity with other predictors.'
Besides their use to decompose explained variance (something I am interested in), using structure coefficients is also a helpful diagnostic when (footnote 1) : Suppression is apparent when a predictor has a beta weight that is disproportionately large (thus receiving predictive credit) relative to a low or near-zero structure coefﬁcient (thus indicating no relationship with the predicted scores).'
Furthermore, when the sum of explained variance of the structure coefficients exceeds 1 (100%) then '... two or more predictors explain some of the same part of the criterion ... suggesting [the presence of] multicollinearity...'
My thinking is that including an option to calculate and report the structure coefficients of a regression model, together with the model coefficients and their beta weight, would be a fine addition to the analytical functionality of `domin`.
Possibly you agree with me on this, but, if not, I am also glad to read your thoughts about the usefulness of structure coefficients.

**Comment by jluchman** [comment]
@ericmelse moved to a discussion as an 'idea'.

I am familiar with structure and commonality analysis (some of this discussion has been in the repository for the R version of `domin` here (https://github.com/jluchman/domir/discussions/9).

In short, it is something I have considered adding for some of the same reasons you mention along with commonality analysis but given the request, I can give it a bit more emphasis. I would then say there's a good chance of adding some of this to `domin` which would probably be a separate command packaged with the primary command. 

Appreciate your thoughts here and passing this on.  Might be a delay in implementing it on the development side but adding it to the list of features to add in future revisions.
